app/uk/gov/hmrc/tradergoodsprofilesrouter/controllers/CreateRecordController.scala

```diff
@@ -45,6 +45,7 @@ class CreateRecordController @Inject() (
   def create(eori: String): Action[JsValue] = authAction(eori).async(parse.json) { implicit request =>
     val result = for {
       _                   <- EitherT.fromEither[Future](validateClientId)
+      _                   <- EitherT.fromEither[Future](validateAcceptHeader)
```

Comment:
just a thought, you may want to put the validate clientID and the validateAccept header in a validateHeader function  and so return a list of error, like we done with validateQueryParameters

Reply:
Currently  we have tickets to remove the `x-client-id` from several endpoints and int he near future we are going to remove it entirely. That's why I kept them as separate methods.